agent/php_execute.c

```diff
@@ -919,40 +919,97 @@ static void nr_php_execute_file(const zend_op_array* op_array,
 }
 
 /*
- * Version specific metadata that we have to gather before we call the original
- * execute_ex handler, as different versions of PHP behave differently in terms
- * of what you can do with the op array after making that call. This structure
- * and the functions immediately below are helpers for
- * nr_php_execute_enabled(), which is the user function execution function.
+ * Purpose : Add Code Level Metrics (CLM) to a metadata structure from
+ * zend_execute_data.
  *
- * In PHP 7, it is possible that the op array will be destroyed if the function
- * being called is a __call() magic method (in which case a trampoline is
- * created and destroyed). We increment the reference counts on the scope and
- * function strings and keep pointers to them in this structure, then release
- * them once we've named the trace node and/or metric (if required).
- *
- * In PHP 5, execute_data->op_array may be set to NULL if we make a subsequent
- * user function call in an exec callback (which occurs before we decide
- * whether to create a metric and/or trace node), so we keep a copy of the
- * pointer here. The op array itself won't be destroyed from under us, as it's
- * owned by the request and not the specific function call (unlike the
- * trampoline case above).
+ * Params  : 1. A pointer to a metadata structure.
+ *           2. The zend_execute_data
  *
- * Note that, while op arrays are theoretically reference counted themselves,
- * we cannot take the simple approach of incrementing that reference count due
- * to not all Zend Engine functions using init_op_array() and
- * destroy_op_array(): one example is that PHP 7 trampoline op arrays are
- * simply emalloc() and efree()'d without even setting the reference count.
- * Therefore we have to be more selective in our approach.
+ * Note    : It is the responsibility of the caller to allocate the metadata
+ *           structure. In general, it's expected that this will be a pointer
+ *           to a stack variable.
  */
-typedef struct {
-#ifdef PHP7
-  zend_string* scope;
-  zend_string* function;
+static void nr_php_execute_metadata_add_code_level_metrics(
+    nr_php_execute_metadata_t* metadata,
+    NR_EXECUTE_PROTO) {
+#if ZEND_MODULE_API_NO < ZEND_7_0_X_API_NO /* PHP7+ */
+  (void)metadata;
+  NR_UNUSED_SPECIALFN;
```

Comment:
Do we need these lines if we're just going to call return without doing anything to them?

Reply:
Yes, it's because we aren't doing anything with metadata and execute_data when PHP is less than 7 that we need to void both of them explicitly for metadata and with the [macro for execute_data ](https://github.com/newrelic/newrelic-php-agent/blob/main/agent/php_newrelic.h#L61)since it can change depending on PHP version; otherwise, the compiler would complain that the vars were not used.

Reply:
That makes sense, thanks for clarifying 👍